### ISSUE TITLE
Slightly ambiguous placeholder name {EXE PATH} in service creation PowerShell example

### ISSUE THREAD

**Issue by fterrani** [dotnet/AspNetCore.Docs#22406]
I find it not immediately clear in the code below that `{EXE PATH}` refers to the path of the **folder containing the executable**. I thought this was the path to the executable file itself, but this triggered the following error:
`Exception calling "SetAccessRule" with "1" argument(s): "No flags can be set. #Parameter name: inheritanceFlags`.

I noticed the different names `{EXE PATH}` and `{EXE FILE PATH}` only much later (it seems inheritance flags `ContainerInherit,ObjectInherit` can only be used on folders).

I think `{EXE PATH}` should be changed for something clearer, like `{EXE DIRECTORY PATH}`, `{EXE FOLDER PATH}`, `{EXE CONTAINING FOLDER PATH}`...

Here is the original piece of code I was refering to:
```
$acl = Get-Acl "{EXE PATH}"
$aclRuleArgs = "{DOMAIN OR COMPUTER NAME\USER}", "Read,Write,ReadAndExecute", "ContainerInherit,ObjectInherit", "None", "Allow"
$accessRule = New-Object System.Security.AccessControl.FileSystemAccessRule($aclRuleArgs)
$acl.SetAccessRule($accessRule)
$acl | Set-Acl "{EXE PATH}"

New-Service -Name {SERVICE NAME} -BinaryPathName "{EXE FILE PATH}" -Credential "{DOMAIN OR COMPUTER NAME\USER}" -Description "{DESCRIPTION}" -DisplayName "{DISPLAY NAME}" -StartupType Automatic
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b0bd8cb8-7e28-8c4d-a550-3aea8f3ab4dd
* Version Independent ID: 4b60d24e-ef05-15e5-10b4-483094081bf3
* Content: [Host ASP.NET Core in a Windows Service](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/windows-service?view=aspnetcore-5.0&tabs=visual-studio#create-a-service)
* Content Source: [aspnetcore/host-and-deploy/windows-service.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/host-and-deploy/windows-service.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @fterrani ... But immediately under the example, the placeholders are defined, including for `{EXE PATH}` ...

> `{EXE PATH}`: Path to the app's folder on the host (for example, `d:\myservice`). Don't include the app's executable in the path. A trailing slash isn't required.

U said ...

> I thought this was the path to the executable file itself

... but the definition of the placeholder even says *not to include that* and includes an example that doesn't have the executable.

Perhaps, I'm misunderstanding your remarks. Can you clarify how the definition is confusing?

**Comment by guardrex** [comment]
It's easy enough to the change ... I'll go ahead. If anything, it will be even clearer. You did say *slightly* :smile:.

**Comment by Rick-Anderson** [comment]
Path is never the folder, it's the path.

**Comment by guardrex** [comment]
Thanks @fterrani ... We'll take more feedback on this. I'm ok with the current form **_WITH_** the existing examples that I placed in the text. That's why I have those there. I knew that it would be a bit tricky without them.

WRT the definition of **_path_**, I think the industry has overloaded the term. Various definitions say it can be both, and the style manual seems to bear that out ...

https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/path

... so let's take more feedback on it. We can look again if devs write in again on this. Thanks again for the shout out, @fterrani. We do need to look when things seem a bit tricky and/or an error occurs when trying to follow the guidance.

**Comment by guardrex** [comment]
... and btw .... I had a bit of a ...

# 😆

... for ...

> When a path also specifies a file, use full path.

*full path*? ... overloading **_absolute path_** (vs. relative path). Making things even more overloaded, maybe? I say so because a folder path could also be absolute (from the root) or relative (to another folder). Anyway .... I gotta get back to work! :smile: lol ... This might be fun to discuss over a few 🍻 sometime.

**Comment by fterrani** [comment]
Sorry for replying only now, my github notifications were lost in the cosmic depths of my inbox... Anyway.

> Perhaps, I'm misunderstanding your remarks. Can you clarify how the definition is confusing?

Here is how I see it: if we consider the whole content of the section (that is the code example and the definition of placeholders right below it), it is indeed very clear and coherent ; I don't think anyone sane can argue with that.

When I used that piece of code, I was in a hurry and since placeholder names seemed pretty self-explanatory at first sight, I copied the code without reading the rest of the section. I genuinely thought `{EXE PATH}` and `{EXE FILE PATH}` were exactly the same thing and that someone might have forgotten to use the same name in both places.

I might be mistaken but to me, a path is a path and it can refer to a folder, or a file, or even other stuff on Linux. It's in the name after all, a path leads somewhere/to something and that somewhere/something could be anything. So consequently, I interpret `{EXE PATH}` as a path of/to `EXE` => `EXE` is an executable file => a path of/to `EXE` is then a path of/to an executable file.

I should have taken the time to read the whole section more carefully. Still, changing `{EXE PATH}` to `{EXE FOLDER PATH}` would make this already pretty clear section crystal-clear and its code easier to use. This isn't a big problem though (hence the word "slightly" and the low number of people in this thread)... I'm aware that I might be nitpicking here, but hey, I'm a programer so that's both my job and my "condition" :) (pun totally intended)
